### PR TITLE
fix(pushnotification): deprecated build.gradle

### LINKED ISSUE #9537: @aws-amplify/pushnotification native code is incompatible with React Native >= 0.65
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

Push Notifications

### Amplify Categories

notifications

### Environment information

<details>

```
 System:
    OS: macOS 10.15.7
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 517.60 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 14.18.1 - ~/.nvm/versions/node/v14.18.1/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v14.18.1/bin/yarn
    npm: 6.14.10 - ~/.nvm/versions/node/v14.18.1/bin/npm
    Watchman: 2021.12.13.00 - /usr/local/bin/watchman
  Browsers:
    Chrome: 97.0.4692.99
    Firefox: 91.5.0
    Safari: 15.3
  npmPackages:
    @aws-amplify/pushnotification: ^4.3.10 => 4.3.10
    @babel/core: ^7.12.9 => 7.16.12
    @babel/runtime: ^7.12.5 => 7.16.7
    @react-native-community/eslint-config: ^2.0.0 => 2.0.0
    HelloWorld:  0.0.1
    aws-amplify: ^4.3.13 => 4.3.13
    babel-jest: ^26.6.3 => 26.6.3
    eslint: 7.14.0 => 7.14.0
    hermes-inspector-msggen:  1.0.0
    jest: ^26.6.3 => 26.6.3
    metro-react-native-babel-preset: ^0.66.0 => 0.66.2
    react: 17.0.2 => 17.0.2
    react-native: 0.65.2 => 0.65.2
    react-native-codegen: ^0.0.7 => 0.0.7
    react-test-renderer: 17.0.2 => 17.0.2
  npmGlobalPackages:
    @aws-amplify/cli: 7.6.3
    detox-cli: 19.0.0
    expo-cli: 5.0.1
    npm: 6.14.10
    wml: 0.0.83
    yarn: 1.22.17

```

</details>

### Describe the bug

Running `yarn android` after installing `aws-amplify` and `@aws-amplify/pushnotification` fails:

```sh
> Configure project :aws-amplify_pushnotification
WARNING:: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'.
It will be removed in version 7.0 of the Android Gradle plugin.
For more information, see http://d.android.com/r/tools/update-dependency-configurations.html.
WARNING:: Please remove usages of `jcenter()` Maven repository from your build scripts and migrate your build to other Maven repositories.
This repository is deprecated and it will be shut down in the future.
See http://developer.android.com/r/tools/jcenter-end-of-service for more information.
Currently detected usages in: project ':aws-amplify_pushnotification'

> Task :app:mergeDebugAssets FAILED
8 actionable tasks: 5 executed, 3 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeDebugAssets'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.firebase:firebase-jobdispatcher:0.6.0.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/com/firebase/firebase-jobdispatcher/0.6.0/firebase-jobdispatcher-0.6.0.pom
       - file:/Users/cpollman/.m2/repository/com/firebase/firebase-jobdispatcher/0.6.0/firebase-jobdispatcher-0.6.0.pom
       - file:/Users/cpollman/Amplify/Issues/JS-9528/Version65/node_modules/react-native/android/com/firebase/firebase-jobdispatcher/0.6.0/firebase-jobdispatcher-0.6.0.pom
       - file:/Users/cpollman/Amplify/Issues/JS-9528/Version65/node_modules/jsc-android/dist/com/firebase/firebase-jobdispatcher/0.6.0/firebase-jobdispatcher-0.6.0.pom
       - https://dl.google.com/dl/android/maven2/com/firebase/firebase-jobdispatcher/0.6.0/firebase-jobdispatcher-0.6.0.pom
       - https://www.jitpack.io/com/firebase/firebase-jobdispatcher/0.6.0/firebase-jobdispatcher-0.6.0.pom
     Required by:
         project :app > project :aws-amplify_pushnotification

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s

error Failed to install the app. Make sure you have the Android development environment set up: https://reactnative.dev/docs/environment-setup.
Error: Command failed: ./gradlew app:installDebug -PreactNativeDevServerPort=8081

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeDebugAssets'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.firebase:firebase-jobdispatcher:0.6.0.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/com/firebase/firebase-jobdispatcher/0.6.0/firebase-jobdispatcher-0.6.0.pom
       - file:/Users/cpollman/.m2/repository/com/firebase/firebase-jobdispatcher/0.6.0/firebase-jobdispatcher-0.6.0.pom
       - file:/Users/cpollman/Amplify/Issues/JS-9528/Version65/node_modules/react-native/android/com/firebase/firebase-jobdispatcher/0.6.0/firebase-jobdispatcher-0.6.0.pom
       - file:/Users/cpollman/Amplify/Issues/JS-9528/Version65/node_modules/jsc-android/dist/com/firebase/firebase-jobdispatcher/0.6.0/firebase-jobdispatcher-0.6.0.pom
       - https://dl.google.com/dl/android/maven2/com/firebase/firebase-jobdispatcher/0.6.0/firebase-jobdispatcher-0.6.0.pom
       - https://www.jitpack.io/com/firebase/firebase-jobdispatcher/0.6.0/firebase-jobdispatcher-0.6.0.pom
     Required by:
         project :app > project :aws-amplify_pushnotification

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s
```

### Expected behavior

Android builds should be successful.

### Reproduction steps

1. run `npx react-native init SampleApp` (React Native v0.67) OR `npx react-native init SampleApp  --version 0.65`
2. run `cd SampleApp`
3. run `yarn add aws-amplify @aws-amplify/pushnotification`
4. run `yarn android`

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### aws-exports.js

_No response_

### Manual configuration

_No response_

### Additional configuration

_No response_

### Mobile Device

_No response_

### Mobile Operating System

_No response_

### Mobile Browser

_No response_

### Mobile Browser Version

_No response_

### Additional information and screenshots

_No response_

### PR DESCRIPTION
**Pull request by Simon-TechForm**
#### Description of changes
Replaces deprecated use of `compile` with `implementation` and removes redundant `firebase-jobdispatcher` dependency in `build.gradle`.

Since `react-native@0.65.0` `firebase-jobdispatcher` is no longer resolved, likely due to the removal of the jcenter repository. See [react native docs](https://reactnative.dev/blog/2021/08/17/version-065#notable-dependency-version-updates-and-gotchas) & #9537.
While looking for a solution, I noticed that I don't see the dependency being used anywhere in the java code? I believe that it's an unused leftover. Please feel free to sanity check me on this. Removing the dependency (obviously) fixes it not being resolved, but also doesn't introduce any other errors.

`react-native@0.68.0` bumps the android gradle plugin to 7.x which removes the `compile` configuration. See: [gradle docs](https://docs.gradle.org/current/userguide/upgrading_version_6.html#sec:configuration_removal) & #9606

#### Issue #, if available
Fixes #9537 & #9606

#### Description of how you validated changes
Implemented the changes in a local `react-native@0.68.0-rc.1` project, which now builds and runs successfully.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by calebpollman** [comment]
@Simon-TechForm We are testing this for backwards compatibility, just wanted to give you context on why this has not been reviewed yet

**Comment by Samaritan1011001** [comment]
@Simon-TechForm Thank you for the PR. Appreciate it a lot. I manually tested this fix's compatibility with RN versions `60, 65 and 68.rc1` and seems to be working fine. I am further investigating writing a few tests.

**Comment by Simon-TechForm** [comment]
@calebpollman @Samaritan1011001 sounds good. Yeah, I had a quick look around in the repo to see if you had any tests like this already, but didn't manage to find any. And I'm not really sure how one would go around automating a test for this.

**Comment by Samaritan1011001** [comment]
Test Details: 

Push notification package version: `"@aws-amplify/pushnotification": "^4.3.11"`

Functionality test criteria: 
- registers the device and provides a device token
- all the callbacks are fired based on the respective events -- `onRegistered`, `onNotification`, `onNotificationOpened`
- notification is received when sent from Pinpoint console
- notification is received when the app is in `Foreground, Background, killed` states
- `onNotificationOpened` is fired when the user taps on the notification in all the app states mentioned above. 

| RN version | Functionality Tested | Build Tested |
| --- | --- | --- |
| 60| ✅  |✅  |
| 61  | ❌  | ✅   |
| 62  | ❌  | ✅   |
| 63 | ❌ | ✅   |
| 64  | ❌ | ✅   |
| 65  | ✅   |✅    |
| 66  | ❌  |✅  |
| 67 | ✅  |✅ |
| 68.rc1 | ✅   |✅  |

**Review by calebpollman — APPROVED** [review]
LGTM, thanks for the contribution @Simon-TechForm and the testing @Samaritan1011001

File: packages/pushnotification/android/build.gradle [hunk 1]
```diff
@@ -48,10 +48,9 @@ android {
...
 dependencies {
-    compile fileTree(dir: "libs", include: ["*.jar"])
-    compile "com.android.support:appcompat-v7:23.0.1"
-    compile "com.facebook.react:react-native:+"  // From node_modules
-    compile 'com.google.firebase:firebase-messaging:23.0.0'
-	compile 'com.google.firebase:firebase-core:20.0.0'
-	compile 'com.firebase:firebase-jobdispatcher:0.6.0'
+    implementation fileTree(dir: "libs", include: ["*.jar"])
+    implementation "com.android.support:appcompat-v7:23.0.1"
+    implementation "com.facebook.react:react-native:+"  // From node_modules
```
Comment (sktimalsina) [inline packages/pushnotification/android/build.gradle:53]: nit: It's good practice to point to a stable version to avoid any foreseen issue.

Comment (Simon-TechForm) [reply packages/pushnotification/android/build.gradle:53]: Actually I'm pretty sure it's the exact opposite in this specific case. You do not want to pin the version, as it should pick up the version installed in node_modules by the library consumer.

Either way, wouldn't changing this be out of the scope of this pr? :)

Comment (sktimalsina) [reply packages/pushnotification/android/build.gradle:53]: I agree. we can look at `+` separately :D

**Review by sktimalsina — APPROVED** [review]

**Comment by Samaritan1011001** [comment]
@Simon-TechForm Thank you for your contribution!

**Comment by Simon-TechForm** [comment]
Hi @mauerbac. I've left you a message on Twitter.

**Pull request merged**